### PR TITLE
Issue#248 exposed methodtorefresh grid

### PR DESCRIPTION
**Pull request by AlexisDragneel**

### CODE REVIEW

File: sample/src/data/ICommonProps.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export default interface ICommonProps {
```
Comment (geoperez) [inline sample/src/data/ICommonProps.ts]: I don't like the name, maybe IDataGridProps

File: sample/src/remoteDataGrid.tsx [hunk 2]
```diff
@@ -8,15 +8,17 @@ import DataGrid, {
   withRemoteDataSource
 } from '../../src';
 import columns from './data/columns';
+import ICommonProps from './data/ICommonProps';
+import IGridState from './IGridState';
```
Comment (geoperez) [inline sample/src/remoteDataGrid.tsx]: Also rename this to IDataGridState please

File: src/DataSource/BaseDataSource.tsx [hunk 3]
```diff
@@ -191,15 +190,14 @@ export default abstract class BaseDataSource extends React.Component<
...
-
     return (
       <DataSourceContext.Provider
         value={{
           actions: this.getActions(),
           state: { ...this.state }
         }}
       >
-        <WrappedComponet error={this.state.error} {...this.props} />
+        <WrappedComponet {...{error:this.state.error, refresh:() => this.retrieveData.bind(this) }}  />
```
Comment (geoperez) [inline src/DataSource/BaseDataSource.tsx]: But if the component has props.children?

Comment (AlexisDragneel) [reply src/DataSource/BaseDataSource.tsx]: the props.children is always present in react

Comment (geoperez) [reply src/DataSource/BaseDataSource.tsx]: 👍

File: sample/src/data/IDataGridProps.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+export default interface IDataGridProps {
```
Comment (geoperez) [inline sample/src/data/IDataGridProps.ts]: why this file is located in data folder?

**Pull request merged**